field_float_nullable/static/src/js/float_field.esm.js

```diff
@@ -0,0 +1,70 @@
+/** @odoo-module **/
+
+import {Component} from "@odoo/owl";
+import {_lt} from "@web/core/l10n/translation";
+import {formatFloat} from "@web/views/fields/formatters";
+import {parseFloat} from "@web/views/fields/parsers";
+import {registry} from "@web/core/registry";
+import {standardFieldProps} from "@web/views/fields/standard_field_props";
+import {useInputField} from "@web/views/fields/input_field_hook";
+import {useNumpadDecimal} from "@web/views/fields/numpad_decimal_hook";
+
+export class FloatNullableField extends Component {
```

Comment:
Is it not better to extend FloatField instead of Component, and call super in parse and get formattedValues to reuse the base logic of the float field. There are examples in Odoo, such as FloatScannableField.